src/config.rs

```diff
@@ -133,4 +150,77 @@ impl Config {
     pub fn parse<T: AsRef<[u8]>>(buf: T) -> Result<Config, ParseError> {
         grammar::parse(buf.as_ref())
     }
+
+    /// Return the suffixes declared in the last "domain" or "search" directive.
+    ///
+    /// ```rust
+    /// # extern crate resolv_conf;
+    /// use resolv_conf::{ScopedIp, Config};
+    /// # fn main() {
+    /// let config_str = "search example.com sub.example.com\ndomain localdomain";
+    /// let parsed_config = Config::parse(&config_str).expect("Failed to parse config");
+    /// let domains = parsed_config.get_last_search_or_domain()
+    ///                            .map(|domain| domain.clone())
+    ///                            .collect::<Vec<String>>();
+    /// assert_eq!(domains, vec![String::from("localdomain")]);
+    ///
+    /// let config_str = "domain localdomain\nsearch example.com sub.example.com";
+    /// let parsed_config = Config::parse(&config_str).expect("Failed to parse config");
+    /// let domains = parsed_config.get_last_search_or_domain()
+    ///                            .map(|domain| domain.clone())
+    ///                            .collect::<Vec<String>>();
+    /// assert_eq!(domains, vec![String::from("example.com"), String::from("sub.example.com")]);
+    /// # }
+    pub fn get_last_search_or_domain<'a>(&'a self) -> DomainIter<'a> {
+        match self.last_search {
+            LastSearch::Search => DomainIter::Search(
+                self.get_search()
+                    .and_then(|domains| Some(domains.into_iter())),
+            ),
+            LastSearch::Domain => DomainIter::Domain(self.get_domain()),
+            LastSearch::None => DomainIter::None,
+        }
+    }
+
+    /// Return the domain declared in the last "domain" directive.
+    pub fn get_domain(&self) -> Option<&String> {
+        self.domain.as_ref()
+    }
+
+    /// Return the domains declared in the last "search" directive.
+    pub fn get_search(&self) -> Option<&Vec<String>> {
+        self.search.as_ref()
+    }
+
+    /// Set the domain corresponding to the "domain" directive.
+    pub fn set_domain(&mut self, domain: String) {
+        self.domain = Some(domain);
+        self.last_search = LastSearch::Domain;
+    }
+
+    /// Set the domains corresponding the "search" directive.
+    pub fn set_search(&mut self, search: Vec<String>) {
+        self.search = Some(search);
+        self.last_search = LastSearch::Search;
+    }
+}
+
+/// A iterator returned by [`Config.get_last_search_or_domain`](struct.Config.html#method.get_last_search_or_domain)
+#[derive(Debug, Clone)]
+pub enum DomainIter<'a> {
```

Comment:
This enum should be hidden behind newtype with a private field (so we don't expose internal structure)

And also should be re-exported in the crate, so that you can name a iterator type returned.

Reply:
done :)